django/core/validators.py

```diff
@@ -449,3 +450,53 @@ def __eq__(self, other):
             self.max_digits == other.max_digits and
             self.decimal_places == other.decimal_places
         )
+
+
+@deconstructible
+class FileExtensionValidator(object):
+    message = _(
+        "File extension '%(extension)s' is not allowed. "
+        "Allowed extensions are: '%(allowed_extensions)s'."
+    )
+    code = 'invalid_extension'
+
+    def __init__(self, allowed_extensions=None, message=None, code=None):
+        self.allowed_extensions = allowed_extensions
+        if message is not None:
+            self.message = message
+        if code is not None:
+            self.code = code
+
+    def __call__(self, value):
+        extension = os.path.splitext(value.name)[1][1:].lower()
+        if self.allowed_extensions is not None and extension not in self.allowed_extensions:
+            raise ValidationError(
+                self.message,
+                code=self.code,
+                params={
+                    'extension': extension,
+                    'allowed_extensions': ', '.join(self.allowed_extensions)
+                }
+            )
+
+    def __eq__(self, other):
```

Comment:
Added tests to tests/validators/tests.py and addressed your other comments. Thanks!